docker/src/main/java/org/jclouds/docker/compute/functions/ContainerToNodeMetadata.java

```diff
@@ -113,8 +112,8 @@ private String cleanUpName(String name) {
 
    private List<String> getPublicIpAddresses(Container container) {
       String dockerIpAddress;
-      if (container.node() != null && !Strings.isNullOrEmpty(container.node().ip())) {
-         dockerIpAddress = container.node().ip();
+      if (container.node() != null && container.node().isPresent()) {
```

Comment:
If using optionals we should get rid of the null check and assume it is never null.